pkgs/development/libraries/gl3w/default.nix

```
@@ -0,0 +1,34 @@
+{ lib, stdenv, fetchFromGitHub, python3, cmake, libglvnd, libGLU }:
+
+stdenv.mkDerivation rec {
+  pname = "gl3w";
+  version = "unstable-2022-03-24";
+
+  src = fetchFromGitHub {
+    owner = "skaslev";
+    repo = pname;
+    rev = "5f8d7fd191ba22ff2b60c1106d7135bb9a335533";
+    hash = "sha256-qV/PZmaP5iCHhIzTA2bE4d1RMB6LzRbTsB5gWVvi9bU=";
+  };
+
+  nativeBuildInputs = [ python3 cmake ];
+  propagatedBuildInputs = [ libglvnd.dev libGLU.dev ];
```

Is this really necessary?

---

This package builds to a header file, a C source file, and a CMake file. When you use that CMake library target in another project, it expects to be able to build the C file and link to both `OpenGL::GL` and `OpenGL::GLU` CMake targets (which is not an ideal way for them to do that but :woman_shrugging:)

If this is not the proper usage of `propagatedBuildInputs` (I understood it to ensure certain other things become a `buildInput` when this package is used as a `buildInput`), then I can remove it. We'll just need to make sure both of those packages are included when including gl3w in anything else

Thank you for the thorough review, by the way!

---

It is a proper use but its not good coding style from upstream.

Would the following work? I am unsure right now.

```suggestion
  propagatedBuildInputs = [ libglvnd libGLU ];
```

---

Yes, that seems to work in my testing. I'll go ahead and update that

---

I went ahead and added a comment explaining the need for that line as well (in case someone stumbles across it later and wonders why that's there). It's definitely not an ideal way for upstream to be handling that though (they're basically patching in an extra C file to your executable when you do `target_link_libraries([...] PRIVATE gl3w)` in CMake)